Added an index node in managers configuration

einorler opened the pull request:
Closes #661

saimaz commented on DependencyInjection/Configuration.php:
```diff
@@ -89,7 +89,6 @@ private function getConnectionsNode()
 
         $node
-            ->isRequired()
             ->requiresAtLeastOneElement()
```
Change to `defautl(null)`

saimaz commented on DependencyInjection/Configuration.php:
```diff
@@ -158,9 +157,53 @@ private function getManagersNode()
                     ->end()
+                    ->arrayNode('index')
+                        ->children()
+                            ->scalarNode('name')
```
change to `index_name`

saimaz commented on DependencyInjection/Configuration.php:
```diff
@@ -158,9 +157,53 @@ private function getManagersNode()
-                        ->isRequired()
                         ->info('Sets connection for manager.')
                     ->end()
+                    ->arrayNode('index')
```
Reuse `getConnectionsNode` by introducing node name parameter.

einorler commented:
Changed `index` node `name` to `index_name`, refactored `process` method of the compiler pass, added a default value to the `connections` node, updated a test

saimaz commented:
@einorler can you update the docs with changes of this PR?

einorler commented:
Sure
